packages/next/client/index.js

```diff
@@ -96,9 +96,28 @@ class Container extends React.Component {
         })
     }
 
+    // call router.replace to trigger data fetching while
+    // the fallback is shown
+
+    if (data.isFallback) {
```

Comment:
Why is this special cased? This should be in the clause below but with `shallow: !data.isFallback` right?

Reply:
Merged it and updated the shallow option